lib/codit/population/networks/home.py

```diff
@@ -0,0 +1,113 @@
+"""
+An spatial attribute for each household
+"""
+
+import pandas as pd
+import os
+import csv
+import numpy as np
+from codit import share_dir
+
+DATA_PATH = os.path.join(share_dir(), 'codit', 'data')
+COORDINATES_CSV = os.path.join(DATA_PATH, 'city', 'population', 'coordinates_leeds.csv')
+TYPES_CONSTRAINTS_CSV = os.path.join(DATA_PATH, 'city', 'population', 'types_households_constraints_leeds.csv')
+COORDINATES_NUM_HOUSEHOLDS_CSV = os.path.join(DATA_PATH, 'city', 'population', 'coordinates_num_households_leeds.csv')
+
+building_types = ["apartments",
+                 "bungalow",
+                 "cabin",
+                 "detached",
+                 "dormitory",
+                 "farm",
+                 "ger",
+                 "hotel",
+                 "house",
+                 "houseboat",
+                 "residential",
+                 "semidetached_house",
+                 "static_caravan",
+                 "terrace"]
+
+class Home:
+    def __init__(self, lon=0.0, lat=0.0, accommodation_type=''):
+        self.coordinate = {'lon': lon, 'lat': lat}
+        self.type = accommodation_type
+
+def get_coords(csvfilename):
+    """
+    Get coordinates and building_type from previous queries to OpenStreetMap
+    :param csvfilename: csv file that stores the coordinates and building_type
+    :return: return list of [float(coord['lon']), float(coord['lat']), str(coord['building_type'])]
+    """
+    coords  = []
+    with open(csvfilename, 'r') as csv_coords_f:
+        coords_rd = csv.DictReader(csv_coords_f)
+        coords += [[float(coord['lon']), float(coord['lat']), str(coord['building_type'])]
+               for coord in coords_rd]
+        return coords
+
+def count_coords_for_types(coords):
+    """
```

Comment:
please could you clarify this docstring?

Reply:
The coordinates list requested from Openstreetmap is organized as a list of ['lon','lat','building_type']. Here I just want to get the number of coordinates for eacy building_type within the coordinates list, e.g. number of houses, number of apartments. I also use 'types_households_constraints_leeds.csv' to assume a minimum number and maximum number of households for each building-type. For example, I assumed the min and max number of households of a house are both 1. The min number of households in an apartment is 20, while max number of households in an apartment is 100. Number of households' distribution in the map should follow the distribution of each type of building. Apartment and terrace coodinates should have higher density of population.

Reply:
This docstring is generated automatically by PyCharm when I input """""", and press enter in the middle of two """. Kind of function descriptions. It will pop up when move cursor to the function name.

Reply:
Sure. Will add it. Thank you.